commit 5df491010117197409a9ae327655bc464a246361
Author: Jeff Law <jeffreyalaw@gmail.com>
Date:   Tue Feb 11 05:47:49 2025 -0700

    Revert "newlib: str[n]cat: optimize skipping of the destination string"
    
    This reverts commit 9a706d0479af5691f14a57a1114215c03c8d3d63.

diff --git a/newlib/libc/string/strcat.c b/newlib/libc/string/strcat.c
index 47c53a5d2..71dd1db75 100644
--- a/newlib/libc/string/strcat.c
+++ b/newlib/libc/string/strcat.c
@@ -50,17 +50,17 @@ strcat (char *__restrict s1,
 #else
   char *s = s1;
 
-  /* Skip unaligned memory in s1.  */
-  while (UNALIGNED_X(s1) && *s1)
-    s1++;
 
-  /* Skip over the aligned data in s1 as quickly as possible.  */
-  unsigned long *aligned_s1 = (unsigned long *)s1;
-  while (!DETECT_NULL(*aligned_s1))
-    aligned_s1++;
-  s1 = (char *)aligned_s1;
+  /* Skip over the data in s1 as quickly as possible.  */
+  if (!UNALIGNED_X(s1))
+    {
+      unsigned long *aligned_s1 = (unsigned long *)s1;
+      while (!DETECT_NULL(*aligned_s1))
+	aligned_s1++;
+
+      s1 = (char *)aligned_s1;
+    }
 
-  /* Find string terminator.  */
   while (*s1)
     s1++;
 
diff --git a/newlib/libc/string/strncat.c b/newlib/libc/string/strncat.c
index fc9fe5b1c..01f20f681 100644
--- a/newlib/libc/string/strncat.c
+++ b/newlib/libc/string/strncat.c
@@ -59,17 +59,16 @@ strncat (char *__restrict s1,
 #else
   char *s = s1;
 
-  /* Skip unaligned memory in s1.  */
-  while (UNALIGNED_X(s1) && *s1)
-    s1++;
+  /* Skip over the data in s1 as quickly as possible.  */
+  if (!UNALIGNED_X(s1))
+    {
+      unsigned long *aligned_s1 = (unsigned long *)s1;
+      while (!DETECT_NULL(*aligned_s1))
+	aligned_s1++;
 
-  /* Skip over the aligned data in s1 as quickly as possible.  */
-  unsigned long *aligned_s1 = (unsigned long *)s1;
-  while (!DETECT_NULL(*aligned_s1))
-    aligned_s1++;
-  s1 = (char *)aligned_s1;
+      s1 = (char *)aligned_s1;
+    }
 
-  /* Find string terminator.  */
   while (*s1)
     s1++;
 
